### PR TITLE
Tighten security around potential P2P issues

### PR DESCRIPTION
**Pull request by sug0**
## Describe your changes

Decrease CometBFT block sizes, and add a new genesis parameter used to validate the max tx bytes allowed for a certain tx to be included in the mempool (passes `CheckTx`, therefore is broadcasted and disseminated over CometBFT's P2P layer) and in `ProcessProposal`.

## Indicate on which release or other PRs this topic is based on

`v0.25.0`

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

File: apps/src/lib/config/genesis/templates.rs [hunk 1]
```diff
@@ -242,6 +242,8 @@ pub struct Parameters<T: TemplateValidation> {
     Eq,
 )]
 pub struct ChainParams<T: TemplateValidation> {
+    /// Max payload size, in bytes, for a mempool tx.
```
Comment (adrianbrink) [inline apps/src/lib/config/genesis/templates.rs:245]: This isn't for the mempool. This is the max size of a tx enforced in consensus

File: core/src/types/chain.rs [hunk 2]
```diff
@@ -130,23 +130,19 @@ impl ProposalBytes {
...
-    /// overhead, evidence and header data, and 90 MiB for
-    /// tx data.
+    /// overhead, evidence and header data. For P2P safety
+    /// reasons (i.e. DoS protection) we hardcap the size of
+    /// tx data to 6 MiB.
     const RAW_MAX: NonZeroU64 = unsafe {
         // SAFETY: We are constructing a greater than zero
         // value, so the API contract is never violated.
-        NonZeroU64::new_unchecked(90 << 20)
+        NonZeroU64::new_unchecked(6 * 1024 * 1024)
```
Comment (adrianbrink) [inline core/src/types/chain.rs:145]: Does this still need to be unsafe?

**Review by adrianbrink — COMMENTED** [review]
Pls double-check the comments I left

**Review by tzemanovic — DISMISSED** [review]

**Review by batconjurer — DISMISSED** [review]

File: core/src/types/chain.rs (hunk 2 above)
Comment (sug0) [reply core/src/types/chain.rs:145]: yes

**Comment by adrianbrink** [comment]
Replaced by #2187

**Pull request closed without merging**